crates/bevy_window/src/event.rs

```diff
@@ -330,3 +330,22 @@ pub struct WindowThemeChanged {
     /// The new system theme.
     pub theme: WindowTheme,
 }
+
+/// Application lifetime events
+#[derive(Event, Debug, Clone, PartialEq, Eq, Reflect)]
+#[reflect(Debug, PartialEq)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+pub enum Lifetime {
```

Comment:
Could we rename this as something like `ApplicationLifetime`, or `AppLifetime`?

If I saw this without knowing what it was, I would initially assume it had something to do with rust lifetimes.